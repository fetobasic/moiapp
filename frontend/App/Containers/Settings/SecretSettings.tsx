import React from 'react';
import { View, Text, StyleSheet, TextStyle, StyleProp } from 'react-native';
import { Colors, Fonts, Metrics } from 'App/Themes';
import { Pressable } from 'App/Components';
import { setToClipboard } from 'App/Services/Clipboard';

export type Row = {
  title: string;
  description?: string;
  style?: StyleProp<TextStyle>;
  iconSettings?: { icon: React.ReactNode; handlePress: () => void };
  useClipboard?: boolean;
};

export type InformationRowProps = {
  rows: Array<Row>;
};

function SecretSettings({ rows }: InformationRowProps) {
  return (
    <View>
      {rows.map(({ title, description, iconSettings, style: textStyle, useClipboard }, ind) => (
        <Pressable
          disabled={!useClipboard}
          onPress={() => setToClipboard(description)}
          style={[styles.container, ind === 0 && styles.firstRow, ind === rows.length - 1 && styles.lastRow]}>
          <View style={[styles.textContainer]}>
            <Text style={[styles.text]} numberOfLines={1} ellipsizeMode="tail">
              {title}
            </Text>
          </View>
          <View style={[styles.rightSideContainer]}>
            {iconSettings ? (
              <Pressable onPress={iconSettings.handlePress} style={styles.btn}>
                {iconSettings.icon}
              </Pressable>
            ) : (
              <Text style={[styles.text, textStyle]}>{description}</Text>
            )}
          </View>
        </Pressable>
      ))}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 16,
    paddingHorizontal: Metrics.halfMargin,
    borderWidth: 1,
    borderColor: Colors.border,
  },
  firstRow: {
    borderTopLeftRadius: 16,
    borderTopRightRadius: 16,
  },
  lastRow: {
    borderBottomLeftRadius: 16,
    borderBottomRightRadius: 16,
  },
  btn: {
    padding: 3,
  },
  textContainer: { flex: 1, justifyContent: 'center' },
  rightSideContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    ...Fonts.font.base.bodyTwo,
  },
});

export default SecretSettings;
