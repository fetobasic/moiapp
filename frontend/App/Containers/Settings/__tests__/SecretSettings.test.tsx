import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState } from 'App/Fixtures/mocks/mockedState';
import SecretSettings from '../SecretSettings';

const mockStore = createMockStore([]);
const store = mockStore({
  ...mockedState,
});

describe('SecretSettings', () => {
  test('renders screen', () => {
    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <SecretSettings
            rows={[
              {
                title: 'Title',
                description: 'Description',
              },
            ]}
          />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
