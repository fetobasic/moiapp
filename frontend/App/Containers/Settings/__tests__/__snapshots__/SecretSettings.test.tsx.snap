// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecretSettings renders screen 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        [
          {
            "borderColor": "rgba(54, 54, 54, 1)",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 12,
            "paddingVertical": 16,
          },
          {
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
          },
          {
            "borderBottomLeftRadius": 16,
            "borderBottomRightRadius": 16,
          },
        ],
      ]
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "color": "rgba(255, 255, 255, 0.87)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(255, 255, 255, 0.87)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            undefined,
          ]
        }
      >
        Description
      </Text>
    </View>
  </View>
</View>
`;
